src/radiation/radiation.cpp

```diff
@@ -0,0 +1,621 @@
+//
+// Created by owen on 3/19/22.
+//
+#include "radiation.hpp"
+#include <fstream>
+#include <iostream>
+#include <set>
+#include <utility>
+#include "environment/runEnvironment.hpp"
+#include "finiteVolume/compressibleFlowFields.hpp"
+#include "utilities/mathUtilities.hpp"
+
+ablate::radiation::RadiationSolver::RadiationSolver(std::string solverId, std::shared_ptr<domain::Region> region, int rayNumber,
+                                                    std::shared_ptr<parameters::Parameters> options)
+    : CellSolver(std::move(solverId), std::move(region),  std::move(options)) {
+    raynumber = rayNumber;
+    nTheta = raynumber; //The DEFAULT number of angles to solve with, should be given by user input probably?
+    nPhi = 2 * raynumber; //The DEFAULT number of angles to solve with, should be given by user input
+}
+
+ablate::radiation::RadiationSolver::~RadiationSolver() {
+}
+
+void ablate::radiation::RadiationSolver::Setup() {  // allows initialization after the subdomain and dm is established
+    ablate::solver::CellSolver::Setup();
+    dim = subDomain->GetDimensions();  // Number of dimensions already defined in the setup
+
+}
+void ablate::radiation::RadiationSolver::Initialize() {
+    // this->reallySolveParallelPlates();
+    this->RayInit();  // Runs the ray initialization finding cell indices
+    /*this->RayTrace(0);
+    for (int i = 1; i < 20; i++) {
+        this->RayProduct(0, i);
+    }*/
+
+}
+
+/// Declaring function for the initialization to call, draws each ray vector and gets all of the cells associated with it (sorted by distance and starting at the boundary working in)
```

Comment:
These comments are really great and something I should do better with.  Would you mind formatting them as [doxygen](https://www.doxygen.nl/manual/docblocks.html) so that they show up on ablate.dev?